posthog/middleware.py

```diff
@@ -145,11 +145,31 @@ def __init__(self, get_response):
 
     def __call__(self, request: HttpRequest):
         if request.user.is_authenticated:
+            path_parts = request.path.strip("/").split("/")
+            project_id_in_url = None
+            if len(path_parts) >= 2 and path_parts[0] == "project" and path_parts[1].isdigit():
+                project_id_in_url = int(path_parts[1])
+            elif (
+                len(path_parts) >= 3
+                and path_parts[0] == "api"
+                and path_parts[1] == "project"
+                and path_parts[2].isdigit()
+            ):
+                project_id_in_url = int(path_parts[2])
+
+            if (
+                project_id_in_url is not None
+                and request.user.team is not None
+                and request.user.team.pk != project_id_in_url
+            ):
+                new_team = request.user.teams.get(pk=project_id_in_url)
```

Comment:
I think this crashes bad if you type a team id that you don't have access to like http://localhost:8000/project/100

Reply:
Maybe we need to wrap this whole thing in a try / except?